src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -634,6 +638,109 @@ public virtual ODataResource CreateResource(SelectExpandNode selectExpandNode, R
             }
         }
 
+        /// <summary>
+        /// Method to append InstanceAnnotations to the ODataResource and Property
+        /// </summary>
+        /// <param name="resource">param for ODataResource</param>
+        /// <param name="resourceContext">param for ODataResourceContext</param>        
+        public virtual void AppendInstanceAnnotations(ODataResource resource, ResourceContext resourceContext)
+        {
+            IEdmStructuredType structuredType = resourceContext.StructuredType;
+            IEdmStructuredObject structuredObject = resourceContext.EdmObject;
+            PropertyInfo instanceAnnotationInfo = EdmLibHelpers.GetInstanceAnnotationsDictionary(structuredType,
+                resourceContext.EdmModel);
+
+            object value;
+            
+            if (instanceAnnotationInfo == null || structuredObject == null ||
+                !structuredObject.TryGetPropertyValue(instanceAnnotationInfo.Name, out value) || value == null)
+            {
+                return;
+            }                  
+
+            IDictionary<string, IDictionary<string, object> > instanceAnnotationsDictionary = value as IDictionary<string, IDictionary<string, object>>;
+
+            if(instanceAnnotationsDictionary != null)
+            {
+                if (instanceAnnotationsDictionary.ContainsKey(string.Empty))
```

Comment:
Use `TryGetValue` here instead so you avoid a duplicate lookup below - it's a waste of CPU